website/content/api-docs/secret/transit.mdx

```diff
@@ -722,8 +728,9 @@ will be returned.
 - `nonce` `(string: "")` – Specifies the **base64 encoded** nonce value. This
   must be provided if convergent encryption is enabled for this key and the key
   was generated with Vault 0.6.1. Not required for keys created in 0.6.2+. The
-  value must be exactly 96 bits (12 bytes) long and the user must ensure that
-  for any given context (and thus, any given encryption key) this nonce value is
+  value must be exactly 96 bits (12 bytes) long for AES and ChaCha20 or 192
+  bits (24 bytes) long for XChaCha20 and the user must ensure that for any
+  given context (and thus, any given encryption key) this nonce value is
```

Comment:
Does this need to be supported for XChaCha20?  I don’t think it is of any value there, since Vault 0.6.1 didn’t support XChaCha20.  Therefore, it just adds a way for users to do something insecure.

Reply:
I do wonder if we can just remove it. Upstream wasn't too happy about that, but given how old `v0.6.x` is, it might be fine to do now for the fork.

I think this accurately reflects the code though. While there is no way to do this via API, if you overwrite storage and fudge the key version there, I think you can force the backend into this state, and so this documentation would be correct if you do that.

Reply:
Filed https://github.com/openbao/openbao/issues/37.